CMakeLists.txt

```diff
@@ -960,10 +960,25 @@ IF (WITH_H2OLOG)
 ENDIF (WITH_H2OLOG)
 
 # gitrev
-IF ("${VERSION_PRERELEASE}" STREQUAL "-DEV" AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
+SET(GITREV_IS_AVAILABLE FALSE)
+IF (EXISTS "${CMAKE_SOURCE_DIR}/.git")
+    # asks generate_gitrev.pl whether or not gitrev is available.
+    EXECUTE_PROCESS(
+        COMMAND perl ${CMAKE_CURRENT_SOURCE_DIR}/misc/generate_gitrev.pl
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        RESULT_VARIABLE GITREV_RESULT_CODE
+        OUTPUT_QUIET
+        ERROR_QUIET
+    )
```

Comment:
Is there a problem in writing gitrev.h here? Unless there's a problem I do not think it is necessary to support two invocation modes in generate_gitrev.pl.

Reply:
Done ed9291cf7cec07438280d7b7471604e35e90a540

Also related to the other review: https://github.com/h2o/h2o/pull/2537/files#r553141379

So now I agreed not to touch generate_gitrev.pl, since it's called in a dev release.